File: packages/csharp/HtmlToMarkdown/Metadata/TableGrid.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace HtmlToMarkdown.Metadata;

/// <summary>
/// A structured table grid with row/column dimensions and cell-level data.
/// </summary>
public record TableGrid
{
    /// <summary>
    /// Number of rows in the table.
    /// </summary>
    [JsonPropertyName("rows")]
    public uint Rows { get; init; }

    /// <summary>
    /// Number of columns in the table.
    /// </summary>
    [JsonPropertyName("cols")]
    public uint Cols { get; init; }

    /// <summary>
    /// All cells in the table (may be fewer than rows*cols due to spans).
    /// </summary>
    [JsonPropertyName("cells")]
    public List<GridCell> Cells { get; init; } = [];
}

File: packages/java/src/main/java/dev/kreuzberg/htmltomarkdown/TableGrid.java
package dev.kreuzberg.htmltomarkdown;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/**
 * A structured table grid with row/column dimensions and cell-level data.
 *
 * @param rows number of rows
 * @param cols number of columns
 * @param cells all cells in the table (may be fewer than rows*cols due to spans)
 */
public record TableGrid(
    @JsonProperty("rows") int rows,
    @JsonProperty("cols") int cols,
    @JsonProperty("cells") List<GridCell> cells) { }
